R/github.R

```diff
@@ -40,114 +30,162 @@ get_github_user <- function(token) {
 #' @param token You can provide the Personal Access Token key directly or this function will attempt to grab a PAT that was stored using the `authorize("github")` function
-#' @param owner Who is the owner of this account? For example in the repository fhdsl/metricminer, fhdsl is the owner
-#' @param repo What is the repository name? For example in the repository fhdsl/metricminer, "metricminer" is the repo name.
+#' @param repo The repository name. So for `https://github.com/fhdsl/metricminer`, it would be `fhdsl/metricminer`
+#' @param count How many items would you like to recieve? Put "all" to retrieve all records.
 #' @return Information regarding a github account
 #' @importFrom gh gh
 #' @export
 #' @examples \dontrun{
 #'
 #' authorize("github")
-#' get_github_user()
+#' metrics <- get_github_metrics(repo = "fhdsl/metricminer")
 #' }
-get_github_metrics <- function(token, owner, repo) {
+get_github_metrics <- function(repo, token = NULL, count = "all") {
+
+  if (count == "all") count <- "Inf"
 
   if (is.null(token)) {
     # Get auth token
     token <- get_token(app_name = "github")
-  } else {
-    token <- token
   }
 
-  community <- gh::gh("GET /repos/{owner}/{repo}/community/profile",
-         owner = owner,
-         repo = repo,
-         .token = token)
+  # Split it up
+  split_it <- strsplit(repo, split = "\\/")
+  owner <- split_it[[1]][1]
+  repo <- split_it[[1]][2]
+
+  api_calls <- list(
+    repo_activity = "GET /repos/{owner}/{repo}/activity",
+    stars = "GET /repos/{owner}/{repo}/stargazers",
+    forks = "GET /repos/{owner}/{repo}/forks",
+    contributors = "GET /repos/{owner}/{repo}/contributors",
+    community = "GET /repos/{owner}/{repo}/community/profile",
+    clones = "GET /repos/{owner}/{repo}/traffic/clones",
+    views = "GET /repos/{owner}/{repo}/traffic/views"
+  )
 
-  clones <- gh::gh("GET /repos/{owner}/{repo}/traffic/clones",
+  results <- lapply(api_calls, function(api_call) {
+    gh_repo_wrapper(api_call = api_call,
                     owner = owner,
                     repo = repo,
-                    .token = token)
+                    token = token,
+                    count = count
+                    )
+  })
+
+  return(results)
+}
+
+
+#' Retrieve metrics for all repos of an organization
+#' @description This is a function to get metrics for all the repos underneath an organization
+#' @param token You can provide the Personal Access Token key directly or this function will attempt to grab a PAT that was stored using the `authorize("github")` function
+#' @param owner The owner of the repository. So for `https://github.com/fhdsl/metricminer`, it would be `fhdsl`
+#' @return Information regarding a github account
+#' @importFrom gh gh
+#' @importFrom purrr map
+#' @export
+#' @examples \dontrun{
+#'
+#' authorize("github")
+#' all_repos_metrics <- get_org_metrics(owner = "fhdsl")
+#' }
+#'
+get_org_metrics <- function(owner = NULL, token = NULL) {
```

Comment:
Again, got rid of default `NULL` for the argument `owner` to force the user to specify that argument (or else the function won't run)

```suggestion
get_org_metrics <- function(owner, token = NULL) {
```